src/views/Register/RegisterPage.tsx

```diff
@@ -0,0 +1,78 @@
+import { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom'
+import { BeatLoader } from 'react-spinners';
+import mapFirebaseErrorMessages from '../../mapFirebaseErrorMessages';
+import { useAuthContext } from '../../context/AuthContext';
+
+
+function RegisterPage() {
+
+  const context = useAuthContext()
+  const navigate = useNavigate()
+  const [loading, setLoading] = useState<boolean>(false)
+  const [error, setError] = useState<string | null>(null)
+
+  const onSubmit = (e: React.FormEvent) => {
+    e.preventDefault()
+    const formData = new FormData(e.currentTarget as HTMLFormElement);
+    const email = formData.get("email") as string
+    const name = formData.get("name") as string
+    const password = formData.get("password") as string
+    const repeatPassword = formData.get("repeatPassword") as string
+
+    if(email === "" || name === "" || password === "" || repeatPassword === "") {
+      setError("Please fill out all the fields.")
+      return
+    }
+    if(repeatPassword !== password) {
+      setError("Passwords don't match.")
+      return
+    }
+
+    setLoading(true)
+    context.register(email, password)
+      .then((userCredential) => {
+        context.saveUserDb(userCredential.user.uid, email, name)
+          .then(() => {
+            //login automatically after register
+            context.login(email, password, true)
+              .then(() => {
+                setLoading(false)
+                navigate("/")
+              })
+              .catch((loginError) => {
+                setLoading(false)
+                setError(mapFirebaseErrorMessages(loginError.code))
+              })
+          })
+          .catch((dbError) => {
+            setLoading(false)
+            setError(mapFirebaseErrorMessages(dbError.code))
+          })
+      })
+      .catch((registerError) => {
+        setLoading(false)
+        setError(mapFirebaseErrorMessages(registerError.code))
+      })
```

Comment:
Using the `Promise.then` method, you can chain these consecutive calls without nesting them, like:

```js
register()
  .then(() => saveToDB())
  .then(() => login())
```